rq/job.py

```diff
@@ -482,7 +488,9 @@ def cancel(self):
 
     def delete(self, pipeline=None):
```

Comment:
I think it would be better to introduce an argument "remove_from_queue" which defaults to True. In "job.cleanup", we would then call  `job.delete(remove_from_queue=False)`